view/frontend/templates/button_product_page.phtml

```diff
@@ -289,23 +289,35 @@ foreach ($block->getAdditionalCheckoutButtonAttributes() as $attrName => $attrVa
                 <?php if ($block->isGrouped()): ?>
                     <?php foreach ($block->getGroupedProductChildren() as $childProduct):?>
                         var childProductQty = getGroupedProductChildQty(<?= /* @noEscape */ $childProduct->getId(); ?>);
+                        <?php if ($childProduct->getTypeInstance()->isVirtual($childProduct)): ?>
```

Comment:
Let's calculate productType only if childProductQty > 0, it increases readability